test_aodndata/moorings/test_mooringsProductsHandler.py

```diff
@@ -25,33 +28,100 @@
         upload_broker.upload(INPUT_FILE_COLLECTION)
 
         handler = self.run_handler(GOOD_MANIFEST)
         self.assertEqual(len(handler.excluded_files), 1)
 
+    def test_aodn_moorings_aggregated(self):
+        expected_prefix = 'IMOS/ANMN/QLD/GBRLSL/aggregated_timeseries'
+        filename = 'IMOS_ANMN-QLD_BZ_20121103_GBRLSL_FV01_CPHL-aggregated-timeseries_END-20140522_C-20191120.nc'
+        testfile = os.path.join(self.tempdir, filename)
+        make_test_file(
+            testfile, {
+                'site_code': 'GBRLSL',
+                'source': 'mooring',
+                'featureType': 'timeSeries'
+            })
+        dest_dir, dest_filename = os.path.split(
+            MooringsProductsHandler.dest_path(testfile))
+        self.assertEqual(dest_dir, expected_prefix)
+        self.assertEqual(dest_filename, filename)
+
+    def test_aodn_moorings_hourly_timeseries_qc(self):
+        expected_prefix = 'IMOS/ANMN/QLD/GBRLSL/hourly_timeseries'
+        filename = 'IMOS_ANMN-QLD_BSTUZ_20071103_GBRLSL_FV02_hourly-timeseries_END-20140523_C-20191010.nc'
+        testfile = os.path.join(self.tempdir, filename)
+        make_test_file(
+            testfile, {
+                'site_code': 'GBRLSL',
+                'source': 'mooring',
+                'featureType': 'timeSeries'
+            })
+        dest_dir, dest_filename = os.path.split(
+            MooringsProductsHandler.dest_path(testfile))
+        self.assertEqual(dest_dir, expected_prefix)
+        self.assertEqual(dest_filename, filename)
+
+    def test_aodn_moorings_hourly_timeseries_nonqc(self):
+        expected_prefix = 'IMOS/ANMN/QLD/GBRLSL/hourly_timeseries'
+        filename = 'IMOS_ANMN-QLD_BSTUZ_20071103_GBRLSL_FV02_hourly-timeseries-including-non-QC_END-20140523_C-20191010.nc'
+        testfile = os.path.join(self.tempdir, filename)
+        make_test_file(
+            testfile, {
+                'site_code': 'GBRLSL',
+                'source': 'mooring',
+                'featureType': 'timeSeries'
+            })
+        dest_dir, dest_filename = os.path.split(
+            MooringsProductsHandler.dest_path(testfile))
+        self.assertEqual(dest_dir, expected_prefix)
+        self.assertEqual(dest_filename, filename)
+
+    def test_aodn_moorings_gridded_timeseries(self):
+        expected_prefix = 'IMOS/ANMN/NRS/NRSROT/gridded_timeseries'
+        filename = 'IMOS_ANMN-NRS_SZ_20081120_NRSROT_FV02_PSAL-gridded-timeseries_END-20190523_C-20191121.nc'
+        testfile = os.path.join(self.tempdir, filename)
+        make_test_file(
+            testfile, {
+                'site_code': 'NRSROT',
+                'source': 'mooring',
+                'featureType': 'timeSeriesProfile'
+            })
+        dest_dir, dest_filename = os.path.split(
+            MooringsProductsHandler.dest_path(testfile))
+        self.assertEqual(dest_dir, expected_prefix)
+        self.assertEqual(dest_filename, filename)
+
```

Comment:
Unfortunately ABOS have requested a different path structure, so we need separate test cases (and additional logic in the classifier) for them. The difference is they don't include the site code in the path, so it's just `IMOS/ABOS/<sub-facility>/aggregated_timeseries/` etc...
Here are some test cases, with their correct dest paths:
```
IMOS/ABOS/DA/aggregated_timeseries/IMOS_ABOS-DA_TZ_20150519_EAC4700_FV01_TEMP-aggregated-timeseries_END-20180422_C-20191216.nc
IMOS/ABOS/SOTS/hourly_timeseries/IMOS_ABOS-SOTS_BSTUZ_20100912_SOTS_FV02_hourly-timeseries_END-20190322_C-20190819.nc
IMOS/ABOS/DA/gridded_timeseries/IMOS_ABOS-DA_TZ_20150519_EAC4700_FV02_TEMP-gridded-timeseries_END-20180422_C-20191216.nc
```